sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/TraceIdLong.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.samplers;
+
+import java.util.Arrays;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+final class TraceIdLong {
+  private static final String ALPHABET = "0123456789abcdef";
+  private static final int ASCII_CHARACTERS = 128;
+  private static final byte[] DECODING = buildDecodingArray();
```

Comment:
These are pretty significant 10 lines of code :P We have a policy for our internal package, I don't see a reason to avoid it at the expense of duplicating state.

Reply:
Because the functions that we would make public look very interesting for users to call unfortunately

Reply:
I agree with the sentiment, but we could rename to `OpenTelemetryEncoding` or something I think which would help? `Utils.checkArgument` also seems pretty interesting, and I think to reduce memory usage / improve performance of String building in various places, I'd expect TemporaryBuffers going there at some point - it seems to me like having to think about the deliciousness of classes in the internal package is limiting / not so scalable as a general practice.

Reply:
>  it seems to me like having to think about the deliciousness of classes in the internal package is limiting / not so scalable as a general practice.

Not sure I understand this

Definitely not completely against the other PR. Was an alternative.